* add ssh action

hjanuschka opened the pull request:
adds an action to execute commands on remote host using ssh.

use-case: i am trying to port some of my deployment scripts, and there i needed to restart a webserver via ssh

samples:

```
ssh(
        host: "dev.januschka.com",
        username: "root",
                port: "322",
        commands: [
                     "ls && true",
                     "ls && false",
                     "ls && true"
                    ]
    )
```

executes all commands in the `commands` array, if one of the commands fails, it stops, and raises a exception.

returns a hash

``` ruby
{ 
:stdout  => "WHOLE STDOUT OF ALL COMMANDS",
:stderr => "WHOLE STDERR"
}
```

hjanuschka committed:
* add ssh action

ssh(
		host: "dev.januschka.com",
		username: "root",
		log: false,
		commands: [
				"date && true",
				"echo 'asdasd' && true"
			  ]
	)

KrauseFx commented on fastlane.gemspec:
```diff
@@ -31,6 +31,8 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'addressable', '~> 2.3.8' # Support for URI templates
   spec.add_dependency 'xcode-install', '~> 1.0.1' # Download new Xcode versions
 
+  spec.add_dependency 'net/ssh', '~> 3.0.2' # add SSH gem
```
Please don't introduce new dependencies. Instead use the `Actions.verify_gem!('cocoapods')` method in your integration. Check out `cocoapods.rb` on how to do that :+1:

KrauseFx commented on lib/fastlane/actions/ssh.rb:
```diff
@@ -0,0 +1,153 @@
+        Helper.log.info "\n########### \n #{stdout} \n###############".magenta if params[:log]
+        Actions.lane_context[SharedValues::SSH_STDOUT_VALUE] = stdout
+        Actions.lane_context[SharedValues::SSH_STDERR_VALUE] = stderr
+        {stdout: Actions.lane_context[SharedValues::SSH_STDOUT_VALUE], stderr: Actions.lane_context[SharedValues::SSH_STDERR_VALUE]}
```
Could you add a return to make it clear that you're returning this value

KrauseFx commented on lib/fastlane/actions/ssh.rb:
```diff
@@ -0,0 +1,153 @@
+      end
+
+      def self.return_value
+        # If you method provides a return value, you can describe here what it does
```
Please adds docs here

KrauseFx commented:
Similar to the other PRs, please add Actions.md, make sure the tests are passing and `rubocop` verification works :+1:

hjanuschka commented:
* rubocop is clean
 * tests pass
 * added entry to Actions.md
 * adopted feedback

KrauseFx commented on docs/Actions.md:
```diff
@@ -1746,6 +1746,21 @@ It can process unit test results if formatted as junit report as shown in [xctes
+  username: "root",
+  commands: [
+    "date && true",
+    "date && true"
```
Maybe provide 2 different examples? Why do you need the `&& true` here?

KrauseFx replied:
Oh because of the return status, right? But isn't date returning the right exit code?

KrauseFx commented on lib/fastlane/actions/ssh.rb:
```diff
@@ -0,0 +1,145 @@
+      end
+
+      def self.details
+        # Optional:
```
Please remove this comment

KrauseFx commented on lib/fastlane/actions/ssh.rb:
```diff
@@ -0,0 +1,145 @@
+      end
+
+      def self.authors
+        # So no one will ever forget your contribution to fastlane :) You are awesome btw!
```
Please remove this comment

KrauseFx commented:
Thanks for updating the PR, however we have a merge conflict here. Could you address the remaining comments before it's ready to be merged

hjanuschka commented:
* merge conflict solved (had to put the documentation block at the end of ###misc)
* added 1  more sample - showing a failing ssh case
* removed usless comments

KrauseFx commented:
This looks great, thanks again for all your work :+1:
